docker-cloud/builds/automated-build.md

```diff
@@ -105,6 +105,21 @@ run an initial test.
     Docker Cloud on every push. Only pushes to branches that are listed as the
     source for one or more tags will trigger a build.
 
+### Environment variables for builds
+
+You can set environment variables used in your build either by declaring them in
```

Comment:
There are rare cases where putting confidential info in your source repo is fine. For instance, if I am one guy who has a private repo for an ssh container, I can store the private key for the container image in my source repo since no one else can read it.

We should make the benefits of each approach clear so that users can decide if build rule env vars make sense for their use cases. The biggest benefit of secrets in source repo is that the secrets are version controlled which makes rotating secrets way easier. Of course the drawback is that if you never amend your history git will preserve your secrets forever for anyone who has read access.

So, the general rule of thumb that all secrets should go into build rules is I think too simple. My opinion is that this space is a good place to make clear the subtleties of secrets but our users should already be aware that env variables declared in hooks are accessible when those hooks are run. That's just how shell scripts work.